scripts/background-scripts/urlbar.js

```diff
@@ -0,0 +1,81 @@
+'use strict'
+
+var placeholders = [];
+var url = 'https://raw.github.com/bwinton/whimsy/gh-pages/urlbar-sayings.txt';
+
+setDefaults([
+  'Where do you want to go today?',
+  'Just type ”google.com“.  You know you’re going to.',
+  'Let’s do this thing!',
+  'Hey, I wonder what we should have for lunch?',
+  'Where to, boss?',
+  'I hear Facebook is nice this time of year…',
+  'You know you can search from here, right?',
+  'Have you thought about trying Private Browsing Mode?',
+  'You are in a maze of twisty web pages, all alike.',
+  'Hi!  My name is Url.'
+]);
+
+function setDefaults(defaults){
+  placeholders = defaults;
+}
+function loadPlaceholders() {
+  //load sayings from url
+  return new Promise((resolve, reject) => {
+    var req = new XMLHttpRequest();
+    req.open("GET", url);
+    req.addEventListener("load", () => {
+      if (req.status === 200){
+        resolve(req.response);
+      } else {
+        reject(Error("Error code: " + req.statusText));
+      }
+    });
+    req.send();
+  });
+}
+function onChange(text, suggest){
+  //get preference setting
+  var getting = browser.storage.sync.get('sayings');
+  getting.then((result)=>{
+    if (result.sayings){
```

Comment:
If the user doesn't specify the sayings, then this will come back as null, and so we won't show the sayings, even though the checkbox is checked…  We should probably add a check for `null` here, like the one in `settings/options.js`